src/implementations/astroport/stableswap.rs

```diff
@@ -0,0 +1,190 @@
+use astroport_core::querier::query_supply;
+use astroport_core::U256;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, Decimal, Env, MessageInfo, Response, StdError, StdResult};
+use cosmwasm_std::{Deps, Uint128};
+use cw_asset::{Asset, AssetInfo, AssetInfoBase, AssetList};
+use delegate::delegate;
+use std::cmp::Ordering;
+
+use crate::traits::Pool;
+use crate::CwDexError;
+use astroport_core::asset::Asset as AstroAsset;
+use astroport_core::asset::AssetInfo as AstroAssetInfo;
+
+use super::base_pool::AstroportBasePool;
+use super::helpers::{compute_current_amp, compute_d, AstroAssetList};
+use super::querier::{query_asset_precision, query_pair_config};
+
+pub const N_COINS: u8 = 2;
```

Comment:
is this always 2 or are there pools live where this is set to 3 or more? If there are I think we should make this part of the AstroportStableSwapPool struct, otherwise we can keep it but no need to expose it as pub

Reply:
cool, I'll just make it private